clientapi/routing/sendevent.go

```diff
@@ -76,6 +81,12 @@ func SendEvent(
 		}
 	}
 
+	// create a mutex for the specific user in the specific room
+	// this avoids a situation where events that are received in quick succession are sent to the roomserver in a jumbled order
+	userID := device.UserID
+	mutex, _ := mutexes.LoadOrStore(roomID+userID, &sync.Mutex{})
+	mutex.(*sync.Mutex).Lock()
+
```

Comment:
Please `defer mutex.(*sync.Mutex).Unlock()` instead of unlocking before each `return` further down.